apps_product_creator/models/product_template.py

```diff
@@ -0,0 +1,15 @@
+# Copyright (C) 2017-Today: Odoo Community Association (OCA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    odoo_module_id = fields.Many2one("odoo.module", "Odoo Module", readonly=True,)
+    technical_name = fields.Char(
+        "Technical Name", related="odoo_module_id.technical_name", store=True,
+    )
+    image_module = fields.Binary(
```

Comment:
The problem that I'm seeing now is that the image in the product won't be updated when updated on GitHub. You should remove this field that is useless, and override `write` method in odoo.module, so that when the image is written, you update the linked (if any) product.